python/packages/autogen-ext/src/autogen_ext/models/openai/_openai_client.py

```diff
@@ -145,13 +145,13 @@ def type_to_role(message: LLMMessage) -> ChatCompletionRole:
         return "tool"
 
 
-def user_message_to_oai(message: UserMessage, prepend_name: bool = False) -> ChatCompletionUserMessageParam:
+def user_message_to_oai(message: UserMessage, prepend_name: bool = False, include_name_in_message: bool = True) -> ChatCompletionUserMessageParam:
     assert_valid_name(message.source)
     if isinstance(message.content, str):
         return ChatCompletionUserMessageParam(
             content=(f"{message.source} said:\n" if prepend_name else "") + message.content,
             role="user",
-            name=message.source,
+            name=message.source if include_name_in_message else None,
```

Comment:
`None` is different from not set in `TypeDict` which is the type of `ChatCompletionUserMessageParam`. So instead of what you are doing here, you first create a `ChatCompletionUserMessageParam(content=..., role...)`, and then set `.name` field if `include_name_in_message` is `True`.